src/main/scala/resource/ExtractableManagedResource.scala

```diff
@@ -46,5 +46,7 @@ trait ExtractableManagedResource[+A] extends ManagedResource[A] {
    *        An either where the left hand side is the currently contained resource unless exceptions, in which case
    *        the right hand side will contain the sequence of throwable encountered.
    */
-  def either: Either[Seq[Throwable], A]
+  def either: ExtractedEither[Seq[Throwable], A]
 }
+
+case class ExtractedEither[+A, +B](either: Either[A, B])
```

Comment:
Is there a way to fix the bug without introducing this container?  I'm hesitant to significantly alter the return type of this guy without moving to `scala.util.Try` and using Java 8's "suppressed exception" mechanisms...